File: ResourceManager/ViewModels/Books/EbookPageViewModel.vb
Imports VBNetCore.Mvvm.Abstractions
Imports VBNetCore.Security
Imports ResourceManager.Configuration
Imports ResourceManager.Helpers
Imports System.Collections.ObjectModel
Imports System.Linq
Imports Microsoft.Extensions.DependencyInjection

Namespace ResourceManager.ViewModels
    Public Class EbookPageViewModel
        Inherits PageViewModel
        Public Sub New()

            PageTitle = "Ebook"
            PageIcon = AssetManager.Instance.GetImage("Ebooks.png")
            PageColor = EbookPageColor
            MyBase.HasPermission = AuthManager.User.IsInPermission(VIEW_EBOOK_PERMISSION)

            Dim tagsContent = Application.ServiceProvider.GetRequiredService(Of TagsContentViewModel)()
            tagsContent.PageColor = PageColor
            tagsContent.EbookColumnWidth = 100

            MenuItems = New ObservableCollection(Of PageContentViewModel) From {
    Application.ServiceProvider.GetRequiredService(Of EbookContentViewModel)(),
    tagsContent,
    Application.ServiceProvider.GetRequiredService(Of EbookOptionsContentViewModel)()
}

            SelectedItem = MenuItems.FirstOrDefault()
        End Sub
    End Class
End Namespace

File: ResourceManager/ViewModels/Books/EbookPageViewModel.cs
using Microsoft.Extensions.DependencyInjection;
using NetCore.Mvvm.Abstractions;
using NetCore.Security;
using ResourceManager.Configuration;
using ResourceManager.Helpers;
using System;
using System.Collections.ObjectModel;
using System.Linq;

namespace ResourceManager.ViewModels
{
    public class EbookPageViewModel : PageViewModel
    {
        public EbookPageViewModel()
        {

            PageTitle = "Ebook";
            PageIcon = AssetManager.Instance.GetImage("Ebooks.png");
            PageColor = Constants.EbookPageColor;
            HasPermission = AuthManager.User.IsInPermission(Constants.VIEW_EBOOK_PERMISSION);

            var tagsContent = App.ServiceProvider.GetRequiredService<TagsContentViewModel>();
            tagsContent.PageColor = PageColor;
            tagsContent.EbookColumnWidth = 100;

            MenuItems = new ObservableCollection<PageContentViewModel>
            {
                App.ServiceProvider.GetRequiredService<EbookContentViewModel>(),
                tagsContent,
                App.ServiceProvider.GetRequiredService<EbookOptionsContentViewModel>(),
            };
            
            SelectedItem = MenuItems.FirstOrDefault();
        }
    }
}
